radix.go

```diff
@@ -348,6 +355,39 @@ var defaultDialOpts = []DialOpt{
 	DialTimeout(10 * time.Second),
 }
 
+func parseRedisURL(urlStr string) (string, []DialOpt) {
+	// do a quick check before we bust out url.Parse, in case that is very
+	// unperformant
+	if !strings.HasPrefix(urlStr, "redis://") {
+		return urlStr, nil
+	}
+
+	u, err := url.Parse(urlStr)
+	if err != nil {
+		return urlStr, nil
+	}
+
+	var opts []DialOpt
+	q := u.Query()
+	if p, ok := u.User.Password(); ok {
+		opts = append(opts, DialAuthPass(p))
+	} else if qpw := q.Get("password"); qpw != "" {
+		opts = append(opts, DialAuthPass(qpw))
+	}
+
+	var dbStr string
+	if u.Path != "" && u.Path != "/" {
+		dbStr = u.Path[1:]
+	} else if dbStr = q.Get("db"); dbStr != "" {
+		// cool
```

Comment:
Nitpick: You could move the assignment into the `else if` and change it to an `else` or even remove the whole `else` by initializing `dbStr` to the result of `q.Get("db")` like this:

```go
dbStr := q.Get("db")
if u.Path != "" && u.Path != "/" {
	dbStr = u.Path[1:]
}
```

Reply:
I was being lazy >_< but you're right I should make it nice